app/data/models/User.py

```diff
@@ -1,32 +1,40 @@
 import re
+import json
 from app.data.repositories import UsersRepository
 
 
 class User:
     # TODO should i have validation methods here or to implement them in another way ?
     # TODO when i create a new user I don't have id, what to do with it
-    def __init__(self, username, email, password, birthDay):
-        self.id = None
+    # TODO crypt the password
+    def __init__(self, id, username, email, password, birthday):
+        self.id = id
         self.username = username
         self.email = email
         self.password = password
-        self.birthDay = birthDay
-    
+        self.birthday = birthday
+
     
-    def __init__(self, *data):
-        self.id = data[0]
-        self.username = data[1]
-        self.email = data[2]
-        self.password = data[3]
-        self.birthDay = data[4]
+    @classmethod
+    def new_user(cls, username, email, password, birthday):
+        return cls(None, username, email, password, birthday)
+
 
+    @classmethod
+    def user(cls, args):
+        return cls(*args)
+
+    # OF COURSE IT DOESN'T WORK....
     def validateEmail(self):
 
-        pattern = r".+@\w+.\w+"
+        pattern = r"\.+@\w+.\w+"
```

Comment:
https://regex101.com/
paste your regex there and read explanation on the right.

(dot) is any character if you want dot you do \.

That regex accepts:
2@2.2
_@_._
a@a@a

It does not accept:
email@edu.com.pl (more then one dot in second part)

Reply:
I checked using that website but seemingly didn't check it correctly